This used to be the taggedfileindex patch, but it's gone.
The remaining part just strips off the tag.

--- ./lib/backend/db3.c.orig	2017-01-19 12:54:16.141112342 +0000
+++ ./lib/backend/db3.c	2017-01-19 12:55:14.587945851 +0000
@@ -1024,6 +1024,9 @@ static int dbt2set(dbiIndex dbi, DBT * d
 	    _DBSWAP(hdrNum);
 	    _DBSWAP(tagNum);
 	}
+	/* remove tagged directory info */
+	if (tagNum.ui & 0x80000000)
+	    tagNum.ui &= 0x0000ffff;
 	set->recs[i].hdrNum = hdrNum.ui;
 	set->recs[i].tagNum = tagNum.ui;
     }
